File: backend/app/models/log.py
from __future__ import annotations

from datetime import UTC, datetime

from sqlalchemy import (
    JSON,
    DateTime,
    Index,
    Integer,
    String,
    Text,
)
from sqlalchemy.orm import Mapped, mapped_column

from app.models import Base


class Log(Base):
    __tablename__ = "log"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    timestamp: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(UTC),
        index=True,
    )
    level: Mapped[str] = mapped_column(String(50), index=True)
    message: Mapped[str] = mapped_column(Text)
    extra: Mapped[dict] = mapped_column(JSON, default=dict)


Index("ix_log_timestamp_level", Log.timestamp, Log.level)

File: go-backend/internal/model/log.go
package model

import (
	"time"

	"gorm.io/datatypes"
)

type Log struct {
	ID        uint      `gorm:"primaryKey;index"`
	Timestamp time.Time `gorm:"index;default:current_timestamp"`
	Level     string    `gorm:"size:50;index"`
	Message   string    `gorm:"type:text"`
	Extra     *datatypes.JSON
}

// TableName 对齐 Python SQLAlchemy 的 __tablename__ = "log"。
func (Log) TableName() string {
	return "log"
}
